src/librbd/internal.cc

```diff
@@ -1690,9 +1681,10 @@ namespace librbd {
     uint64_t original_size;
     {
       RWLock::RLocker l(ictx->md_lock);
+      ictx->snap_lock.get_read();
```

Comment:
I don't think it's really necessary anymore. It was held for every call to invalidate_cache(), but I don't see any reason for it.